traffic_ops/traffic_ops_golang/deliveryservice/letsencryptcert.go

```diff
@@ -355,7 +355,10 @@ func GetLetsEncryptCertificates(cfg *config.Config, req tc.DeliveryServiceLetsEn
 	}
 	keyPem := keyBuf.Bytes()
 
-	dsSSLKeys.Certificate = tc.DeliveryServiceSSLKeysCertificate{Crt: string(EncodePEMToLegacyPerlRiakFormat(certificates.Certificate)), Key: string(EncodePEMToLegacyPerlRiakFormat(keyPem)), CSR: ""}
+	// remove extra line if LE returns it
+	trimmedCert := bytes.ReplaceAll(certificates.Certificate, []byte("\n\n"), []byte("\n"))
```

Comment:
Any reason to ever expect a Microsoft style carriage return that would also need to be filtered/removed?

Also, the `EncodePEMToLegacyPerlRiakFormat` (hilarious comments BTW) reintroduces newline characters. Any chance they could end up doubled up again? Or maybe we don't need newlines at all before being passed into `EncodePEMToLegacyPerlRiakFormat`?

Reply:
since this only interacts with Let's Encrypt i wouldnt expect it to ever be different unless they change on their side. so i dont think we need to account for those but i guess it also wouldnt hurt

I think `EncodePEMToLegacyPerlRiakFormat` only adds new line characters between every 76 chars and at the very end so that shouldnt impact this. this is fixing because the LE certs had a new line in the middle of the cert chain that was causing issues.  at least i havent seen any issues with it in testing

also i with i could take credit for the funny comments haha. not from me but i really appreciate them

Reply:
@mattjackson220 So I can see the (_obvious_) fix, but how do I go about testing this? Is there someway for me to generate a LE cert and verify the newlines before and after?